site/content/entry/yuzu-progress-report-aug-2021/index.md

```diff
@@ -0,0 +1,356 @@
+never hurts.
+
+Tired of having to wait over 3 minutes for each change done to the texture cache section of the code, [yzct12345](https://github.com/yzct12345) 
+[split out the definitions](https://github.com/yuzu-emu/yuzu/pull/6820), reducing the build time to 30 seconds.
+
+yzct12345 also found a [deadlock](https://en.wikipedia.org/wiki/Deadlock) in our Single-Producer, Single Consumer queue, 
+[and submitted a fix](https://github.com/yuzu-emu/yuzu/pull/6868) to address this problem.
+The work to rewrite this queue and make it Multi-Producer, Multi-Consumer has also been started, so hopefully we might see a follow-up next month.
+If you’re confused about these "[Producer-Consumer](https://en.wikipedia.org/wiki/Producer%E2%80%93consumer_problem)" terms, just know that they basically describe how to tackle 
+the access to a resource in a multi-core system, so that the processes that write into (produce), and those that read from (consume), a shared resource, are properly synchronized — 
+in this case, the queue.
+
+This adventurous developer also [optimized the UnswizzleTexture function](https://github.com/yuzu-emu/yuzu/pull/6861), yielding a sweet speed gain.
+Swizzling refers to a technique used to optimize how textures are stored in memory to minimize cache misses.
+The opposite operation, thus, would be to take a swizzled texture and reorganize it so it makes sense to humans.
+
+Some time ago, toast detected a potential out-of-bounds access, which is a bug that occurs when the program accesses memory that is outside the range where it should be operating 
+at that given moment.
+The code in question is also part of the function used to unswizzle textures.
+epicboy noticed that the root of the problem laid in the calculation of the frame buffer size, used to limit where the code should operate, 
+[and fixed it on this PR](https://github.com/yuzu-emu/yuzu/pull/6879), effectively eliminating the bug.
+
+While inspecting yuzu manually and with the help of analysis tools, v1993 also found a number of small bugs in different parts of our codebase, such as a 
+[misplaced break statement](https://github.com/yuzu-emu/yuzu/pull/6884) in the kernel function `GetThreadContext()` — a human error that changed the logic of the algorithm, 
+preventing it from behaving as intended.
+In a similar vein, he also corrected a [copy-paste error](https://github.com/yuzu-emu/yuzu/pull/6889) affecting the code of the software keyboard.
+
+v1993 also found another bug in the `GetSubmappedRange()` function, used to obtain the CPU memory segments from a GPU memory address, Blinkhawk 
+[went ahead and fixed it](https://github.com/yuzu-emu/yuzu/pull/6894).
+
+Another noteworthy change by blink is that he [changed the logic of the Garbage Collector](https://github.com/yuzu-emu/yuzu/pull/6897), so that it uses a Least-Recently Used 
```

Comment:
```suggestion
Another noteworthy change by Blinkhawk is that he [changed the logic of the Garbage Collector](https://github.com/yuzu-emu/yuzu/pull/6897), so that it uses a Least-Recently Used 
```